### ISSUE TITLE
New Heath Dogon data

### ISSUE THREAD

**Issue by IndianaTones** [languageorphans/heathdogon#8]
@LinguList please see the file I just uploaded HeathDogonNew.csv. The column BEFORE_PARSE has the original data before our the scripts that Promise wrote to implement our parsings were run, and the column PARSED has data that are ready for the orthography profile.

**Comment by LinguList** [comment]
Can we please add to folder `raw`, where we always start from? I already started to modify the orthoprofile in the morning ;-)

**Comment by IndianaTones** [comment]
Done. It is now called data2.csv and is in the Raw folder.

**Comment by LinguList** [comment]
Thanks!

**Comment by LinguList** [comment]
The code to parse, could you please also upload this to the raw folder, plus the original-original data you use to create the data2.csv? I'd like to start from the original dogon spreasheet, not the derived one, as this is more sustainable if we change more things in the code. I'd include the parsing code into the lexibank script that runs the conversion.

**Comment by LinguList** [comment]
Where does data1.csv come from? Is it based on the CLDF version of HeathDogon? How was it created, what was the manual work? 

I am a bit afraid we are getting into muddy waters if we have too many manual steps with unclear origins of data now, and I cannot directly trace data1.csv to the cldf data, or can I do it in any way?

**Comment by IndianaTones** [comment]
I created data1.csv based on the CLDF version of HeathDogon.

The manual work was separating variants and plurals (everything following the slash that we originally ignored).

**Comment by LinguList** [comment]
But the CLDF version is essentially in the file [cldf/forms.csv](https://github.com/languageorphans/heathdogon/blob/main/cldf/forms.csv), yet this version looks like a LingPy file, so it is different from the CLDF itself, with cognate sets, etc. 

What we must do now is check if we can trace back these forms to the original spreadsheet. In CLDF we can do that, but in this file, I am not sure, as we lack the CLDF identifiers. 

So I wonder how the file was created now: did you use the lingpy export that I provided, and did this include all of the data, or did you use another method? 

So the crucial step CLDF -> new data is not clear to me. 

I can try and find it out by comparing entries, but it would help me to have more information here, how the file was created in the end (e.g., VARID is a value I do not right understand, etc.).

**Comment by IndianaTones** [comment]
I see, you are right. I used [lexibank_heathdogon.py](https://github.com/languageorphans/heathdogon/blob/main/lexibank_heathdogon.py) to generate CONCEPTS with CLDFBENCH and COGID and COGIDS with LingPy.

The VARID we created by hand when we made new rows for each of the phonetic or semantic variants.

**Comment by LinguList** [comment]
How did you use lingpy on the CLDF data, is there a script?

VARID means what? Can you summarize the semantics of the file, what the columns refer to? I cannot see that from looking at entries, of course.

**Comment by IndianaTones** [comment]
`$ cldfbench lexibank.makecldf lexibank_heathdogon.py --glottolog=/Users/linguist/Documents/GitHub/glottolog --glottolog-version=v4.8 --concepticon=/Users/linguist/Documents/GitHub/concepticon-data --concepticon-version=v3.0.0 --clts=/Users/linguist/Documents/GitHub/clts --clts-version=v2.2.0`

**Comment by IndianaTones** [comment]
First, I ran: 

`$ cldfbench lexibank.makecldf lexibank_heathdogon.py --concepticon=MYPATH/concepticon-data --clts=MYPATH/clts --glottolog=MYPATH/glottolog --concepticon-version=v2.5.0 --glottolog-version=v4.3`

This produces HeathDogon.tsv. 

And then I ran the LingPy code to produce COGID and COGIDS, though we’ve only used these thus far for sorting purposes. 

We took HeathDogon and then worked on it collectively in Google Sheets to make manual adjustments.

I realize also in the process that the wordlist was reduced, so it is possible that I ran an additional step to only include Swadesh concepts but for that I am not sure what. 

Very much apologies for the confusion and for not documenting my part of the workflow better but @PromiseDodzi has copius notes as to what we accomplished by hand.

**Comment by LinguList** [comment]
Hm, then you must have used an older version, since the version that we labled version 1.0 did not produce any file HeathDogon.tsv. It produces the CLDF data in the folder `cldf/`. 

But I have now already managed to link your data back to the original Dogon data, I will share results on that soon. I will also propose how to handle plural forms, and I will discuss all these details in the Oberseminar mid March, giving illustrations of how to proceed from there.

**Comment by IndianaTones** [comment]
Yes, I need to keep track of what I do each time. Promise is writing up everything he did as well.

**Comment by LinguList** [comment]
And manual working should only be done, when carefully planned. We discussed plurals, and we could've included them by modifying the code, but now it means my new code is a huge workaround again to integrate data that cannot be completely traced to the CLDF.

**Comment by LinguList** [comment]
So in the data.csv, you have the following cases where we have no form.

```
INFO    No form found in line 2944 (ID: 2,673)
INFO    No form found in line 3713 (ID: 3,372)
INFO    No form found in line 6708 (ID: 6,090)
INFO    No form found in line 9225 (ID: 8,218)
INFO    No form found in line 9466 (ID: 8,432)
INFO    No form found in line 9837 (ID: 8,750)
INFO    No form found in line 9906 (ID: 8,815)
INFO    No form found in line 10586 (ID: 9,411)
INFO    No form found in line 10683 (ID: 9,504)
INFO    No form found in line 11087 (ID: 9,849)
INFO    No form found in line 11198 (ID: 9,945)
INFO    No form found in line 11282 (ID: 10,020)
INFO    No form found in line 11285 (ID: 10,022)
INFO    No form found in line 11286 (ID: 10,022)
INFO    No form found in line 11289 (ID: 10,024)
INFO    No form found in line 11879 (ID: 10,541)
INFO    No form found in line 12731 (ID: 11,246)
INFO    No form found in line 13146 (ID: 11,604)
INFO    No form found in line 13236 (ID: 11,682)
INFO    No form found in line 14570 (ID: 12,919)
INFO    No form found in line 15513 (ID: 13,784)
INFO    No form found in line 15514 (ID: 13,784)
```

Neither singular nor the normal form.

**Comment by LinguList** [comment]
The form `kògò_àr-ⁿá` is most likely wrongly segmented. It should be `kògò_àrⁿ-á`, according to how we read the superscript nasal. And the form `nùmɔ̀-ànà-p̌y` has the little haček-like symbol on top of the `p` not the `y`.

**Comment by IndianaTones** [comment]
> So in the data.csv, you have the following cases where we have no form.

Data.csv is the raw data. We are working with Data1.csv.

**Comment by IndianaTones** [comment]
> The form `kògò_àr-ⁿá` is most likely wrongly segmented. It should be `kògò_àrⁿ-á`, according to how we read the superscript nasal.

You are right. 

> And the form `nùmɔ̀-ànà-p̌y` has the little haček-like symbol on top of the `p` not the `y`.

Yes, this is also a mistake - the tone should be on the `y`.

**Comment by IndianaTones** [comment]
> And manual working should only be done, when carefully planned. We discussed plurals, and we could've included them by modifying the code, but now it means my new code is a huge workaround again to integrate data that cannot be completely traced to the CLDF.

Ok, what can we do now?

**Comment by LinguList** [comment]
I make proposals on May 16 how to proceed, using the code that is now already written by me, that integrates the manual work and offers to make use of plural forms to determine stems.

**Comment by IndianaTones** [comment]
Also please note that, in addition to plurals, we also extracted variants and inflected verb forms by hand. 

Did you find a solution to our stuck vowels in the process?

**Comment by LinguList** [comment]
Yes, the variants are also there now, but please note also, that variants were already available in CLDF, but we decided to take the first form by then. It is just a code switch that we did not do.

**Comment by LinguList** [comment]
The solutions to the vowels is to modify the profile, which I am doing now. I did not check the code that you wrote there for this purpose, but stick to our original profile, since it is easier to start from there, and the code has several different files plus one with 500 lines that I cannot promise to be able to digest until in two weeks, but I also do not think it is necessary.

**Comment by IndianaTones** [comment]
Agreed. Not necessary to go over the new code; it mostly addresses our parsings. Would you want to share with us the new profile though before our meeting so that we can familiarize ourselves already with the process and test that we know how to implement it?

**Comment by LinguList** [comment]
It is already there in the branch [new-profile](https://github.com/languageorphans/heathdogon/tree/new-profile). I will merge before the meeting.

**Comment by IndianaTones** [comment]
Just looked at the new profile - you'll explain to us how it will work with the long vowels I imagine but just to be clear, we want to represent all long vowels as two separate segments and not as aː or àː for example.

**Comment by LinguList** [comment]
You haven't really explained so far, why you want to do this. I generally think it is a bad idea to stretch vowels. But the solution I propose addresses this question in a novel way, why at the same time allowing you to experiment with streched vowels, see our preprint on the topic here: 

> [List, J.-M., N. Hill, F. Blum, and C. Juárez (2024): Grouping sounds into evolving units for the purpose of historical language comparison. Open Research Europe 4.34. 1-8. [Preprint, under review, not peer-reviewed]](https://doi.org/10.12688/openreseurope.16839.1)

**Comment by IndianaTones** [comment]
The stretched vowels are for two reasons:

1) We want to ensure that the 'lost' vowel in cognates are accounted for in the alignments.

2) Our parsings sometimes necessitate that a hyphen be put in between two vowels..

The HeathDogon definition '(e.g. bird) fly' has examples which illustrates both issues:

| Doculect  | Tokens | Alignment |
| ------------- | ------------- |------------- |
| TommoSo  | kííl+í+yé  | k *íí* l + í + y é  |
| DonnoSo  | kílè+è | k í l è + è  |

Note the 'stuck vowels' in the TommoSo root. These roots were not determined to be cognate. We would prefer an alignment such that the loss (or gain) of the additional vowel is reflected in the alignments:

| Doculect  | Tokens | Alignment |
| ------------- | ------------- |------------- |
| TommoSo  | kííl+í+yé  | k *í* *í* l + í + y é  |
| DonnoSo  | kílè+è | k í - l + è + - è |

I do see in your pre-print that your new method takes care of issues such as 'missing' segments, but, Promise created code that correctly parses off inflectional morphology such as the mediopassive here, that in the case of Donno So, is realized as a long vowel. Would your suggested new methods be able to handle such cases?

**Comment by LinguList** [comment]
It is not a method per se, but a representation that preserves the unity of subsequences and allows for more flexibility to select the items one wants to align. In this representation, you have two vowels `i.i` in correspondence with one vowel in another language `i`. If the single vowel has one vowel lost or the other case has one vowel gained is a question of interpretation. The information is the same in all cases, but the advantage is that the new representation can be easily converted into the one with a gap for a vowel, or it could be reversed. In any case, the new orthography profile by now models data in two ways, as grouped sounds and as ungrouped sounds. One can create cognates from both representations. My bet is that alignments will worsen if you use ungrouped sounds. This is based on experience, but also on the fact that vowel sequences are often restricted, and if it is a long vowel, you would not want an alignment to insert a gap between both vowels, which can perfectly happen (`a - a`), but there's no algorithm that would prevent this so far.

**Comment by LinguList** [comment]
I have checked sound correspondences in Dogon now in more cases, and the correspondences are typically very boring (but tone correspondences can be found as well, if we treat tone as a feature of vowels). I think you should -- and we have our little seminar for that as well -- first carry out more hands-on studies on cognates and correspondence patterns, using different data representations, before you make a decision. The new profile gives you essentially all possibilities for that.

**Comment by IndianaTones** [comment]
> It is not a method per se, but a representation that preserves the unity of subsequences and allows for more flexibility to select the items one wants to align. In this representation, you have two vowels `i.i` in correspondence with one vowel in another language `i`. If the single vowel has one vowel lost or the other case has one vowel gained is a question of interpretation. The information is the same in all cases, but the advantage is that the new representation can be easily converted into the one with a gap for a vowel, or it could be reversed. In any case, the new orthography profile by now models data in two ways, as grouped sounds and as ungrouped sounds. One can create cognates from both representations. My bet is that alignments will worsen if you use ungrouped sounds. This is based on experience, but also on the fact that vowel sequences are often restricted, and if it is a long vowel, you would not want an alignment to insert a gap between both vowels, which can perfectly happen (`a - a`), but there's no algorithm that would prevent this so far.

This sounds really exciting and worth trying, but will it interfere with our parsings?

**Comment by IndianaTones** [comment]
> I have checked sound correspondences in Dogon now in more cases, and the correspondences are typically very boring (but tone correspondences can be found as well, if we treat tone as a feature of vowels). I think you should -- and we have our little seminar for that as well -- first carry out more hands-on studies on cognates and correspondence patterns, using different data representations, before you make a decision. The new profile gives you essentially all possibilities for that.

Here is another question we wanted to ask you. We sometimes cannot find the sound correspondences we expect to in Edictor. I think that I am missing something in how to visualize this. For example, concept SHEEP has correspondences [l ~ j ~ z ~ d ~ s] within one cognate set but it doesn't appear in the correspondence patterns for *d reflexes.

**Comment by LinguList** [comment]
I would need a screenshot and an explicit file with explicit alignments where you actually check correspondence patterns now. The question is also, from which language you start inspecting correspondences, and how many you use, and if the correspondence occurs in more than one example, of course!

**Comment by IndianaTones** [comment]
I will get back to you soon with the correspondences issue, but here I made a movie to show you that even with Edictor, our vowels really wish to be stuck together - watch what happens after I manually separate the [ii] in TommoSoTongoTongo and then press Align:

https://github.com/languageorphans/heathdogon/assets/12430221/425a5098-7ec4-4644-9ea8-a557439d3a9a

**Comment by LinguList** [comment]
No, this is another issue, @IndianaTones. It is not that there is a magical feature clustering vowels. It is that you modify the ALIGNMENT manually, but you must modify the column TOKENS.

**Comment by LinguList** [comment]
The columns TOKENS and ALIGNMENT are linked in EDICTOR. If you modify TOKENS, this will be transferred to ALIGNMENT, which makes perfect sense, since you cannot keep the unchanged alignment. 

If you share the file here (zip it and upload in this thread), I can verify that this is no feature but the fact that you did not modify the words, but only the alignment.

**Comment by IndianaTones** [comment]
> No, this is another issue, @IndianaTones. It is not that there is a magical feature clustering vowels. It is that you modify the ALIGNMENT manually, but you must modify the column TOKENS.

In fact @LinguList this is the issue that we wish to address. We cannot prevent these vowels from clustering.

**Comment by LinguList** [comment]
Can you please share the data? I cannot replicate the error if I do not have exactly the same TSV file that you are using there.

**Comment by IndianaTones** [comment]
Sorry, looks like we are speaking in circles. We can discuss it all more clearly tomorrow I believe at this point :)

**Comment by LinguList** [comment]
But uploading a file, so that I can have a look at the very data and the very problem you talk about, can you PLEASE do that before?

**Comment by LinguList** [comment]
A video does not help here for me to debug. I would also need that for tomorrow, otherwise we will loose important time.

**Comment by IndianaTones** [comment]
Replied by email.
